terraform/modules/consumer/lambdas/get_events.py

```diff
@@ -26,26 +26,48 @@
 
 
 def lambda_handler(event, _context):
+    """
+    The implementation here is deliberately not optimal. For example the page size is very small to test pagination
+    and create load on the system for testing purposes. This is designed to verify the system, rather than be a model
+    implementation.
+    """
     logger.info(f"## EVENT: {event}")
     logger.info(f"## TIME: {datetime.now()}")
 
     auth_token = get_auth_token(auth_url, client_id, client_secret)
     start = time.time()
-    events = get_events(auth_token)
+    result_page = get_first_events_page(auth_token)
     stop = time.time()
     delta = stop - start
 
+    total_events = result_page["meta"]["page"]["totalElements"]
+    data = result_page["data"]
+    event_ids = [event['id'] for event in data]
+
+    while "next" in result_page["links"]:
+        result_page = get_data(auth_token, result_page["links"]["next"])
+        page_ids = [event["id"] for event in result_page["data"]]
+        event_ids.extend(page_ids)
+
     record_metric(cloudwatch, cloudwatch_namespace, "GET_EVENTS.Duration", delta, unit="Seconds")
-    record_metric(cloudwatch, cloudwatch_namespace, "GET_EVENTS.Count", len(events))
+    record_metric(cloudwatch, cloudwatch_namespace, "GET_EVENTS.Count", total_events)
 
-    logger.info(f"Received {len(events)} events in {delta}s")
+    logger.info(f"Received {len(event_ids)} events in {delta}s")
+    if total_events != len(event_ids):
+        logger.error(f"Expected {total_events} events but received {len(event_ids)}")
```

Comment:
This is possible because of the nature of limit-offset pagination, but it's worth keeping an eye on